src/nvim/globals.h

```diff
@@ -556,7 +556,6 @@ EXTERN win_T    *prevwin INIT(= NULL);  /* previous window */
   FOR_ALL_TABS(tp) \
     FOR_ALL_WINDOWS_IN_TAB(wp, tp)
 
-# define FOR_ALL_WINDOWS(wp) for (wp = firstwin; wp != NULL; wp = wp->w_next)
```

Comment:
Why was `FOR_ALL_WINDOWS` removed here, `and `FOR_ALL_WINDOWS_IN_TAB(wp, curtab)` used instead?
It causes unnecessary differences, and appears to be the root of warnings with gcc-9:
```
../src/nvim/window.c: In function ‘do_window’:
../src/nvim/window.c:183:33: error: self-comparison always evaluates to true [-Werror=tautological-compare]
  183 |     FOR_ALL_WINDOWS_IN_TAB(wp2, curtab) {
      |                                 ^~
```
(since it compares curtab to curtab then)

Reply:
It's been over two years since I did this.  I don't remember the reasoning.  My guess would be that it seemed like unnecessary duplication.  `FOR_ALL_WINDOWS_IN_TAB` covers all cases that `FOR_ALL_WINDOWS` does.

The only "problem" is that the compiler warns about the comparison, since it doesn't know about the macro.